core/src/test/java/com/graphhopper/routing/weighting/CustomWeightingTest.java

```diff
@@ -0,0 +1,190 @@
+    }
+
+    @Test
+    public void testSpeedFactorBooleanEV() {
+        CustomModel vehicleModel = new CustomModel();
+
+        BooleanEncodedValue rcLinkEnc = encodingManager.getBooleanEncodedValue(RoadClassLink.KEY);
+        Map map = new HashMap<>();
+        map.put("true", 0.5);
+        vehicleModel.getPriority().put(RoadClassLink.KEY, map);
+        CustomWeighting weighting = new CustomWeighting(carFE, encodingManager, NO_TURN_COST_PROVIDER, vehicleModel);
+
+        assertEquals(3.1, weighting.calcEdgeWeight(graphHopperStorage.edge(0, 1).setDistance(10).
+                set(rcLinkEnc, false).set(avSpeedEnc, 15).set(accessEnc, true), false), 0.01);
+        assertEquals(5.5, weighting.calcEdgeWeight(graphHopperStorage.edge(0, 1).setDistance(10).
+                set(rcLinkEnc, true).set(avSpeedEnc, 15).set(accessEnc, true), false), 0.01);
+    }
+
+    @Test
+    public void testPriority() {
+        CustomModel vehicleModel = new CustomModel();
+
+        Map map = new HashMap();
+        map.put(MOTORWAY.toString(), 0.1);
+        vehicleModel.getPriority().put(KEY, map);
+        CustomWeighting weighting = new CustomWeighting(carFE, encodingManager, NO_TURN_COST_PROVIDER, vehicleModel);
+
+        // simple multiplication of the weight...
+        assertEquals(24.7, weighting.calcEdgeWeight(graphHopperStorage.edge(0, 1).setDistance(10).
+                set(roadClassEnc, MOTORWAY).set(avSpeedEnc, 15).set(accessEnc, true).setReverse(accessEnc, true), false), 0.01);
+        assertEquals(3.1, weighting.calcEdgeWeight(graphHopperStorage.edge(0, 1).setDistance(10).
+                set(roadClassEnc, PRIMARY).set(avSpeedEnc, 15).set(accessEnc, true).setReverse(accessEnc, true), false), 0.01);
+        assertEquals(123.5, weighting.calcEdgeWeight(graphHopperStorage.edge(0, 1).setDistance(50).
+                set(roadClassEnc, MOTORWAY).set(avSpeedEnc, 15).set(accessEnc, true).setReverse(accessEnc, true), false), 0.01);
+        assertEquals(15.5, weighting.calcEdgeWeight(graphHopperStorage.edge(0, 1).setDistance(50).
+                set(roadClassEnc, PRIMARY).set(avSpeedEnc, 15).set(accessEnc, true).setReverse(accessEnc, true), false), 0.01);
+    }
+
+    @Test
+    public void testAvoidArea() {
+        CustomModel vehicleModel = new CustomModel();
+
+        Map map = new HashMap();
+        map.put(MOTORWAY.toString(), 0.1);
+        String areaId = "blup";
+        vehicleModel.getPriority().put(KEY, map);
+        vehicleModel.getPriority().put("area_" + areaId, 0.5);
+
+        Coordinate[] coordinates = {new Coordinate(13.722, 51.053), new Coordinate(13.722, 51.055),
+                new Coordinate(13.731, 51.055), new Coordinate(13.731, 51.053), null};
+        coordinates[coordinates.length - 1] = coordinates[0];
+        Geometry poly = new GeometryFactory().createPolygon(coordinates);
+
+        JsonFeature area = new JsonFeature(areaId, "Polygon", new BBox(13.713684, 13.719864, 51.036213, 51.036591),
```

Comment:
Why do we have to pass an extra bounding box here, when we already specified the polygon?

Reply:
The JSON feature allows for a separate bounding box definition that I used (long before we had JTS) to either use the BBox in it or the geometry. But in this specific case we can probably give it null.

Reply:
Ah I see.